### PR TITLE
flysystem to fetch calendars, contacts and profile for a user

### PR DESCRIPTION
**Pull request by ylebre**

### CODE REVIEW

File: src/NextcloudCalendar.php [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+use OCA\DAV\Connector\LegacyDAVACL;
+use OCA\DAV\CalDAV\CalendarRoot;
+use OCA\DAV\Connector\Sabre\Auth;
+use OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin;
+use OCA\DAV\Connector\Sabre\MaintenancePlugin;
+use OCA\DAV\Connector\Sabre\Principal;
+
+
+/**
+ * Filesystem adapter to convert RDF files to and from a default format
```
Comment (Potherca) [inline src/NextcloudCalendar.php]: Copy/Paste error?

File: src/NextcloudCalendar.php [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+{
+    private $defaultAcl;
+    private $userId;
+    private $principalUri;
+
+    final public function __construct($userId, $defaultAcl)
+    {
+        $this->userId = $userId;
+        $this->principalUri = "principals/users/" . $this->userId;
+	$this->defaultAcl = $defaultAcl;
```
Comment (Potherca) [inline src/NextcloudCalendar.php]: You might want to check the tabs/spaces in this file.

File: src/NextcloudCalendar.php [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+            \OC::$server->getConfig(),
+            'principals/'
+        );
+        $db = \OC::$server->getDatabaseConnection();
+        $userManager = \OC::$server->getUserManager();
+        $random = \OC::$server->getSecureRandom();
+        $logger = \OC::$server->getLogger();
+        $dispatcher = \OC::$server->getEventDispatcher();
+
+        $this->calDavBackend = new CalDavBackend($db, $principalBackend, $userManager, \OC::$server->getGroupManager(), $random, $logger, $dispatcher, true);
```
Comment (Potherca) [inline src/NextcloudCalendar.php]: For now this is fine but I think the `CalDavBackend` creation will eventually have to be moved outside of the class.

File: src/NextcloudCalendar.php [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+    /**
+     * Read a file as a stream.
+     *
+     * @param string $path
+     *
+     * @return array|false
+     */
+    final public function readStream($path)
+    {
+        return false;
```
Comment (Potherca) [inline src/NextcloudCalendar.php]: In other implementations where no stream is supported, we've just called `return $this->read($path);` here...
Is that a good idea?  :shrug:

File: src/NextcloudCalendar.php [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+     *
+     * @param string $path
+     * @param string $contents
+     * @param Config $config Config object
+     *
+     * @return array|false false on failure file meta data on success
+     */
+    final public function write($path, $contents, Config $config)
+    {
+        $result = true;
```
Comment (Potherca) [inline src/NextcloudCalendar.php]: This seems unused.

**Review by Potherca — APPROVED** [review]
I'm sad I didn't have the energy to build this, but your implementation seems spot on! Great work!

File: src/NextcloudCalendar.php (hunk 2 above)
Comment (ylebre) [reply src/NextcloudCalendar.php]: fixed, converted all to spaces to comply with PSR

File: src/NextcloudCalendar.php (hunk 4 above)
Comment (ylebre) [reply src/NextcloudCalendar.php]: agreed!

File: src/NextcloudCalendar.php (hunk 5 above)
Comment (ylebre) [reply src/NextcloudCalendar.php]: removed that line.

File: src/NextcloudCalendar.php (hunk 1 above)
Comment (ylebre) [reply src/NextcloudCalendar.php]: fixed!

**Comment by Potherca** [comment]
@ylebre I've made the changes we discussed in person. Besides adding support for streams (using the polyfill provided by FlySystem) my changes also:

 - Add missing items to doc-blocks
 - Change all quotes to the same type
 - Fix typo(s)
 - Make implicit declarations, returns and comparisons explicit
 - Move item check to normalize method
 - Move path splitting into a separate method
 - Remove variable assignment in favor of directly returning a value
 - Sort lists alphabetically

**Pull request merged**